roles/openstack-setup/tasks/images.yml

```diff
@@ -1,7 +1,12 @@
 ---
+# NOTE: Glance does not know how to proxy copy-from URLs, so we should instead
+# download the image files locally and create images using the file method.
+- name: download images locally
+  get_url: url={{ item.url }} dest=/tmp/{{ item.filename }}
+  with_items: glance.images
```

Comment:
Does this only run if the glance image is missing? Or does the image get downloaded on every run, and *then* checked against if it already exists in glance?